### DISCUSSION TITLE
DLNA Sources

### DISCUSSION THREAD

**Discussion by omayhemo** [music-assistant/support#890] [Feature requests and ideas]
DLNA sources should be available to Music Assistant.

**Comment by jimothyr** [comment]
this would be super helpful. Home assistant can see my DLNA sources served from a twonky server through the DNLA integration, this would be great to pull into music assistant.

**Reply by damienclaveau** [reply]
+1

**Comment by whealsie** [comment]
I agree. DLNA would be very much better than connecting to the filesystem and bypasses a number of security headaches that connecting to a NAS or file system would expose.

**Comment by solarjunkev** [comment]
100% no brainer. Kindly add DLNA and uPNP sources.

**Comment by busywait** [comment]
Would it be even better, or equally good, if the Music Assistant media players were exposed as DNLA players, and then I could send a playlist of tracks from a UPnP server directly to the player with any DNLA controller?

**Reply by peter--s** [reply]
Well, something similar is possible using the MA integration (or even without, if the desired sink is available in HA already) and the DLNA server integration. All sinks can be exposed to HA and DLNA sources including playlists can be played back. I would still welcome this feature in MA.

**Reply by philanted** [reply]
Can you describe how this works please? HA can 'see' the DLNA Server (Minimserver) but MA cannot. Even with HA/MA integration.

**Reply by peter--s** [reply]
@philanted (and @busywait): Sorry if my comment mislead you (and for the late reply), but what I meant is that the MA integration exposese sinks to HA, so you can play media available to HA (including DLNA) on sinks otherwise only available to MA.
This will not enable DLNA servers to access non-DLNA sinks directly.
Nor can the MA integration make DLNA sources available to MA.

**Comment by solarjunkev** [comment]
Well sure. Anything but the current arrangement. Here we have  music assistant dependent on basically mounting a usb drive/share  within HA. We all know that is incredibly hard to do, Yes it is achievable ..sure..but how many hoops do you have to go thru to simply get a usb drive to mount in linux? It isn't easy. And it risks screwing up your HA code and setup. So we have HA media player that can play individual tracks via DLNA ..but fails due to rubmentary DLNA control. No gapless. No album play. No art or library management etc. And we have the reverse with music assistant - all those goodies but no easy way to get the music library stream via DLNA. And no way to easily mount a share. One option has great input (DLNA). One option has great player (galpless, album art, fade to next track, automatic play to next track etc). But zero options with both. So we are still left without a decent and practical music playback in HA.... still. To this day., Just MO

**Comment by solarjunkev** [comment]
Ok so how long before we can use DLNA sources. Could we have a time frame please? thanks!

**Reply by OzGav** [reply]
Sorry no time frame at the moment. Native SMB support should be coming in the next few weeks

**Comment by johndfoss** [comment]
Is this still being worked on? Thanks!

**Reply by OzGav** [reply]
No sorry this is not being worked on.

**Reply by watje999** [reply]
all my music is only accesible via dlna, there wil never be a file system share for this at my place......
just run minidlna on your linux (ubuntu) server where media resides, to get all your audio,movies, and pictures exposed to some good players at home. (VLC on google tv e.g.) Homeassistant Music assistant could be very well used for indexing, selecting, and called in automations....

**Reply by omayhemo** [reply]
very dogmatic

**Reply by solarjunkev** [reply]
Such a shame DLNA support is "not being worked on"
Oh well...
There goes my dream of using Music Assistant as a full feature music player in Home Assistant!

**Comment by madbrain76** [comment]
This would be very helpful for me also, as  I manage and tag my local collection in Jriver media center, and it can expose it as a DLNA media server.

**Comment by GrantDG** [comment]
Been a while since anyone commented on this option - but I'm going to bump it again.

I'm also a (long time!) JRiver Media Center user and am disappointed that MA doesn't support either of it's options, namely: the proprietary API and/or DLNA.

DLNA seems a great option for MA

**Comment by Steamerzone** [comment]
I was surprised to find out that something like this isn't included into MA, would be a great option to have.

**Comment by GioF71** [comment]
Hello, I believe this will never be available in MA...
DLNA does not prescribe how the media server should expose all its albums.
See e.g. MinimServer. You can get to the same album in a multitude of ways.
Scanning a DLNA media server might never end...

From what I understand, MA needs to acquire contents from a music provider. In the case of streaming services, it "acquires" just the favorites, then allow to search anything via the global search.

**Comment by accma2** [comment]
HA supports DLNA server but MA doesn't...

Don't understand.

**Reply by GioF71** [reply]
In fact HA lets you just _browse_ a DLNA server. MA needs to scan its Music Providers, and a DLNA server isn't good as a Music Provider because it can't be reliably scanned in order to evaluate its entire content. AFAIK there is no standard way to list the complete contents of a DNLA server.
But I'd let to the answer to the MA experts.

**Reply by OzGav** [reply]
The lack of a standard and the many higher priority items means this is not currently being worked on. The lack of a standard cannot be understated. The development effort required to support a wide range of "possible" implementations is considerable. It might even be preferable to have a provider per DLNA server flavour...? The core team does not have capacity to look at this in the medium term.

**Reply by peter--s** [reply]
@OzGav: are you aware of a solution to expose MA sinks via DLNA (as @busywait suggested above, similar to exposing them to HA using the MA integration) or are there ongoing actions to achieve that? It does sound like far less effort.

**Reply by OzGav** [reply]
We aren't looking at that and to be honest given the poor standardisation of the DLNA protocol this won't be high on the agenda.

**Reply by krvest** [reply]
Media player in Windows easily works with miniDLNA showing all details, why MA can't do it. miniDLNA is open source and terefore my first choice. MA without support for DLNA provider is useless.

**Comment by maisun** [comment]
Agree this is very much needed feature - why implement a music lib when there is already one? For example Plex and Jellyfin all can be integrated, what's the issue with DLNA? Not sure I understand..

**Reply by OzGav** [reply]
I answered this immediately above.

**Reply by maisun** [reply]
> needed feature - why implement a music lib when there is already one? For example Plex and Jellyfin all can be integrated, what's the issue with DLNA? Not sure I understand..

Ok I wish emby integration can be added in the future. Now I mount the remote share but for some reason it it super slow when I browser the music.

**Comment by FlatTV** [comment]
Push…

could you please check your agenda and assign a higher priority to this feature?
Sorry - I’ve just added this project and am a bit confused, as DLNA sources are missing, almost „all“ other sources are available.
This is a particularly significant for users who do not use Home Assistant.

**Reply by OzGav** [reply]
Why has DLNA sources got anything to do with Home Assistant?

**Comment by watje999** [comment]
totally agree, it should be possible to browse a minidlna server. That would be awesome to see music and audio collections coming in homeassistant.
